source/common/router/router.cc

```diff
@@ -691,18 +691,20 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
 
 std::unique_ptr<GenericConnPool>
 Filter::createConnPool(Upstream::ThreadLocalCluster& thread_local_cluster) {
-  GenericConnPoolFactory* factory = nullptr;
+  GenericConnPoolFactory* factory = config_.router_context_.genericConnPoolFactory();
+
   if (cluster_->upstreamConfig().has_value()) {
-    factory = Envoy::Config::Utility::getFactory<GenericConnPoolFactory>(
+    auto custom_factory = Envoy::Config::Utility::getFactory<GenericConnPoolFactory>(
```

Comment:
can we revert the code pattern to the previous state? I think the only thing you need to be changing here is how to set the factory if it was not set previously on line 703 in the old code.